File: src/my_project/easy_problems/from101to150/split_string_balanced_string.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        answer: int = 0
        count_r: int = 0
        count_l: int = 0

        for word in s: 
            if word == "R":
                count_r += 1
            else: 
                count_l += 1

            if count_l == count_r:
                answer += 1
        return answer 


solution = Solution()

print(solution.balancedStringSplit( s = "RLRRRLLRLL"))

File: my_project_typescript/easy_problems_typescript/from1to50/splitStringBalancedString.ts
function balancedStringSplit(s: string): number {

    let answer: number = 0
    let countR: number = 0
    let countL: number = 0

    for (let word of s){
        if (word === "R"){
            countR += 1
        }
        else {
            countL += 1
        }

        if (countL === countR){
            answer += 1
        }
    }
    return answer 
};

console.log(balancedStringSplit("RLRRLLRLRL"))

console.log(balancedStringSplit("RLLLLRRRLR"))

console.log(balancedStringSplit("LLLLRRRR"))


console.log(balancedStringSplit("RLRRRLLRLL"))
